src/Core/Domain/ApiAccess/Command/AddApiAccessCommand.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\ApiAccess\Command;
+
+class AddApiAccessCommand
+{
+    public function __construct(
+        public ?string $clientName,
+        public ?string $apiClientId,
+        public ?bool $enabled,
+        public ?string $description
```

Comment:
```suggestion
        public string $clientName,
        public string $apiClientId,
        public bool $enabled,
        public string $description
```
These are required fields so null value should not be allowed If they were then they wouldn't have their place in the constructor Usually we follow a convention for creation command, all required fields are in the constructor so there's no way around it they must be defined And optional fields can be specified by setters

But in this case I believe all those fields are mandatory no? Except maybe enabled which could be false by default for example